faiss/python/__init__.py

```diff
@@ -96,6 +96,21 @@ def replacement_decode(self, codes):
 handle_Quantizer(ScalarQuantizer)
 
 
+def handle_NSG(the_class):
+
+    def replacement_build(self, x, graph):
+        n, d = x.shape
+        graph = graph.reshape((-1,))
```

Comment:
You can assert that graph.shape[0] = n and set K = graph.shape[1]
It does not make sense to accept graphs in arbitrary shapes.